src/pages/spawn/spawnHost/fields/HostExpirationField.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Body } from "@leafygreen-ui/typography";
+import { set } from "date-fns";
+import DatePicker from "components/DatePicker";
+import { InputLabel } from "components/styles";
+import TimePicker from "components/TimePicker";
+
+export interface ExpirationDateType {
+  expiration?: Date;
+  noExpiration?: boolean;
+}
+
+interface HostExpirationFieldProps {
+  data: ExpirationDateType;
+  onChange: React.Dispatch<React.SetStateAction<any>>;
+}
+export const HostExpirationField: React.FC<HostExpirationFieldProps> = ({
+  onChange,
+  data,
+}) => {
+  const { expiration: expirationString, noExpiration } = data;
+  const expiration = new Date(expirationString);
+  const updateDate = (d: Date) => {
+    const year = d.getFullYear();
```

Comment:
I think you need null handling in both places. I'm not exactly sure what setup I used to cause this, but I got a runtime error when d is undefined by clicking the x to clear the date

Reply:
thanks i was able to replicate this I removed the option to clear the date entirely the expiration should be specified by default or no expiration should be set.